src/core/common/settings.hpp

```diff
@@ -784,7 +784,7 @@ class SettingsBase : public InstanceLocator
          * Initializes the `BorderAgentId` object.
          *
          */
-        void Init(void) { mId = {}; }
+        void Init(void) { mId = MeshCoP::BorderAgent::Id(); }
```

Comment:
Does this clear the `mId` array in all cases?

I think the ` MeshCoP::BorderAgent::Id` does not define any specific constructors?
It is actually a C definition as an `struct`
```c++
OT_TOOL_PACKED_BEGIN
struct otBorderAgentId
{
    uint8_t mId[OT_BORDER_AGENT_ID_LENGTH];
} OT_TOOL_PACKED_END;
```

----

I am old-school and would prefer `memset()` to be sure 😄

Reply:
Thanks! Updated. I just wanted to make arm-gcc-4 fine with this.